snet_cli/commands.py

```diff
@@ -504,3 +510,247 @@ def transact(self):
             return receipt, events
         else:
             self._error("Cancelled")
+
+
+class ServiceCommand(BlockchainCommand):
+
+    # We are sorting files before we add them to the .tar since an archive containing the same files in a different
+    # order will produce a different content hash; sorting order is the same default sorting order used in GNU tar.
+    def _tar_imports(self, paths):
+        max_depth = 0
+        deepest_path = PurePath()
+        for path in paths:
+            path_depth = len(path.parts)
+            if not max_depth or path_depth < max_depth:
+                max_depth = path_depth
+                deepest_path = path
+
+        paths = sorted(paths, key=lambda unsorted_path: (len(unsorted_path.parts), str(unsorted_path.name)))
+        f = tempfile.NamedTemporaryFile()
+        tar = tarfile.open(fileobj=f, mode="w")
+        for path in paths:
+            tar.add(path, path.name) if path == deepest_path else tar.add(path, path.relative_to(deepest_path.parent))
```

Comment:
simple solution, the shortest path is already encoded in the array after its sorted AND the path to be relative to is the path that is already passed in to you...

Reply:
Fixed